src/clj/rems/api/applications_v2.clj

```diff
@@ -0,0 +1,869 @@
+                       :textcontent (:default (or (:license/link license)
+                                                  (:license/text license)))
+                       :localizations (into {} (for [lang (-> (set (concat (keys (:license/title license))
+                                                                           (keys (:license/link license))
+                                                                           (keys (:license/text license))))
+                                                              (disj :default))]
+                                                 [lang {:title (get-in license [:license/title lang])
+                                                        :textcontent (or (get-in license [:license/link lang])
+                                                                         (get-in license [:license/text lang]))}]))})
+                    (:application/licenses application))
+     :items (map (fn [field]
+                   {:id (:field/id field)
+                    :type (name (:field/type field))
+                    :optional (:field/optional field)
+                    :options (:field/options field)
+                    :maxlength (:field/max-length field)
+                    :value (:field/value field)
+                    :previous-value nil ; TODO
+                    :localizations (into {} (for [lang (set (concat (keys (:field/title field))
+                                                                    (keys (:field/placeholder field))))]
+                                              [lang {:title (get-in field [:field/title lang])
+                                                     :inputprompt (get-in field [:field/placeholder lang])}]))})
+                 (:form/fields application))}))
+
+(defn api-get-application-v1 [user-id application-id]
+  (when-let [v2 (api-get-application-v2 user-id application-id)]
+    (transform-v2-to-v1 v2 user-id)))
+
+;;; v2 API, listing all applications
+
+(defn- applications-view
+  "Projection for the current state of all applications."
+  [applications event]
+  (if-let [app-id (:application/id event)] ; old style events don't have :application/id
+    (update applications app-id application-view event)
+    applications))
+
+(defn- reset-applications-state [_]
+  {:last-processed-event-id 0
+   :applications {}})
+
+(def ^:private applications-state
+  "Stateful projection tracking the current state of all applications.
+  Will be updated periodically by `applications-update-scheduler`,
+  but an update may also be triggered with `trigger-applications-update!`
+  e.g. after committing some new events."
+  (agent (reset-applications-state nil)
+         :error-handler (fn [_agent exception]
+                          (log/error exception "Updating projection failed"))))
+
+(declare trigger-applications-update!)
+
+(defn- update-applications! [state]
+  (let [from-id (:last-processed-event-id state)
+        batch-size 1000
```

Comment:
I'm assuming conman returns all rows eagerly instead of using cursors and returning them lazily, so batching is there to protect from returning 1M rows from database. Though probably all the data fits in memory and batching won't be needed. #863 should investigate that how much data there is to be expected production.

Reply:
We should expect a few thousand applications with a dozen events each so YAGNI?

Reply:
Ok. Removing batching.